lib/closex/mock_client/mock_client.ex

```diff
@@ -187,40 +209,186 @@ defmodule Closex.MockClient do
+  def update_opportunity(@not_found_id, _payload, _opts) do
+    {:error, :mock_not_found}
+  end
+  def update_opportunity(opportunity_id, payload, opts) do
     opportunity = load("opportunity.json")
-    |> Map.merge(payload)
+                  |> Map.merge(payload)
+                  |> Map.merge(%{"id" => opportunity_id})
     send self(), {:closex_mock_client, :update_opportunity, [opportunity_id, payload, opts]}
     {:ok, opportunity}
   end
-  def update_opportunity(@not_found_id, _payload, _opts) do
-    {:error, :mock_not_found}
+
+  @doc """
+  Sends an email CloseIO.
+
+  Returns `{:ok, email}`
+
+  ## Examples
+  
+    > Closex.MockClient.send_email(%{})
+    ...contents of test/fixtures/send_email.json...
+  """
+  def send_email(payload, opts \\ []) do
+    email = load("send_email.json")
+            |> Map.merge(payload)
+    send self(), {:closex_mock_client, :send_email, [payload, opts]}
+    {:ok, email}
   end
 
-  def send_email(payload, opts \\ []), do: Closex.HTTPClient.send_email(payload, opts)
-  def find_leads(search_term, opts \\ []), do: Closex.HTTPClient.find_leads(search_term, opts)
+  @doc """
+  Finds a lead in CloseIO.
+
+  Returns `{:ok, leads}`.
+
+  You can hand in any search term you like and it will return an example search which finds one example lead.
+
+  You can also search for something you are not expecting to find with our not_found_id. This will provide the empty
+  result set in find_no_leads.json.
+
+  We have provided an example search term to use when the search term doesn't matter to you.
+
+  ## Examples
+
+    > Closex.MockClient.update_opportunity(Closex.MockClient.search_term())
+    ...contents of test/fixtures/find_leads.json...
+
+    > Closex.MockClient.update_opportunity(Closex.MockClient.not_found_id())
+    ...contents of test/fixtures/find_no_leads.json...
+  """
+  def find_leads(search_term, opts \\ [])
+  def find_leads(search_term = @not_found_id, opts) do
+    leads = load("find_no_leads.json")
+    send self(), {:closex_mock_client, :find_leads, [search_term, opts]}
+    {:ok, leads}
+  end
+  def find_leads(search_term, opts) do
+    leads = load("find_leads.json")
+    send self(), {:closex_mock_client, :find_leads, [search_term, opts]}
+    {:ok, leads}
+  end
 
   defp load(filename) do
-    fixtures_path = Application.get_env(:closex, :mock_client_fixtures_dir, Path.join([File.cwd!, "lib", "closex", "mock_client", "fixtures"]))
-    [fixtures_path, filename]
-    |> Path.join
+    case Application.fetch_env(:closex, :mock_client_fixtures_dir) do
+      {:ok, fixtures_path} ->
+        file = [fixtures_path, filename]
+               |> Path.join
+               |> File.read
+        case file do
+          {:ok, binary} ->
+            Poison.decode!(binary)
+          {:error, _ } ->
+            load_default(filename)
+        end
+      :error ->
+        load_default(filename)
+    end
+  end
+
+  defp parse_dates_to_strings(lead) do
```

Comment:
Interesting... is this necessary because JSON serialization occurs on DateTimes under the hood automatically?

Reply:
```
> map
%{foo: ~D[2017-01-01]}
> Poison.encode!(map)
"{\"foo\":\"2017-01-01\"}"
```
Yes, it does!

We have to do this translation though because we never get Date or Datetime objects straight out of CloseIO as a response, we always get strings. So we can't just merge the users values into the response and pass it back.